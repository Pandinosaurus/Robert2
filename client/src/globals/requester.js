import axios from 'axios';
import deepObjectSet from 'lodash/set';
import isPlainObject from 'lodash/isPlainObject';
import flattenObject from '@/utils/flattenObject';
import config from '@/globals/config';
import cookies from '@/utils/cookies';
import DateTime from '@/utils/datetime';
import Day from '@/utils/day';

const requester = axios.create({
    baseURL: config.api.url,
    headers: config.api.headers,
});

requester.interceptors.request.use(
    (_request) => {
        const { params, onProgress, ...request } = _request;

        // - Ajoute une méthode haut-niveau permettant de récupérer l'avancement sous forme numérique.
        //   - Si la méthode de la requête est POST, PUT ou PATCH, c'est l'événement `onUploadProgress` qui sera écouté.
        //   - Sinon ce sera `onDownloadProgress`
        if (onProgress) {
            const progressCallback = (event) => {
                if (!event.lengthComputable) {
                    return;
                }

                const { loaded, total } = event;
                const percent = (loaded / total) * 100;
                onProgress(percent);
            };

            if (request.method && ['POST', 'PUT', 'PATCH'].includes(request.method.toUpperCase())) {
                request.onUploadProgress = progressCallback;
            } else {
                request.onDownloadProgress = progressCallback;
            }
        }

        // - Traitement spécial des objets littéraux contenant des fichiers.
        //   => On utilise un envoi via `multipart/form-data` et on met les données
        //      supplémentaires éventuelles dans une clé spéciale `@data`)
        const hasFiles = (
            isPlainObject(request.data) &&
            Object.values(request.data).some((value) => (
                value instanceof File
            ))
        );
        if (hasFiles) {
            const formData = new FormData();
            const data = Object.entries(flattenObject(request.data)).reduce(
                (_data, [key, value]) => {
                    if (value instanceof File) {
                        formData.append(btoa(key), value);
                        return _data;
                    }
                    return deepObjectSet(_data, key, value);
                },
                {},
            );

            formData.append('@data', JSON.stringify(data));
            request.data = formData;
        }

        if (params) {
            Object.keys(params).forEach((name) => {
                if (params[name] === true) {
                    params[name] = '1';
                }
                if (params[name] === false) {
                    params[name] = '0';
                }
                if (params[name] instanceof DateTime || params[name] instanceof Day) {
                    params[name] = params[name].toString();
                }
            });
            request.params = params;
        }

        const authToken = cookies.get(config.auth.cookie);
        if (authToken) {
            request.headers.Authorization = `Bearer ${authToken}`;
        }
        return request;
    },
    (error) => Promise.reject(error),
);

export default requester;
