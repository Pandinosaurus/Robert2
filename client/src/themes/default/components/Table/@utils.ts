import type { ColumnsVisibility } from 'vue-tables-2-premium';

export type ColumnsDisplay = Record<string, boolean>;

const STORAGE_KEY_PREFIX = 'vuetables_';

const initFromStorage = (tableName: string, columns: ColumnsDisplay): ColumnsDisplay => {
    const storedTableState = localStorage.getItem(`${STORAGE_KEY_PREFIX}${tableName}`);
    if (!storedTableState) {
        return columns;
    }

    let tableState;
    try {
        tableState = JSON.parse(storedTableState);
    } catch {
        localStorage.removeItem(`${STORAGE_KEY_PREFIX}${tableName}`);
        return columns;
    }

    if (!tableState) {
        return columns;
    }

    const { userColumnsDisplay } = tableState;
    if (!userColumnsDisplay) {
        return columns;
    }

    return Object.fromEntries(
        Object.keys(columns).map((column: string) => (
            [column, userColumnsDisplay.includes(column)]
        )),
    );
};

export const initColumnsDisplay = (tableName: string | undefined, columns: ColumnsDisplay): ColumnsVisibility => (
    Object.fromEntries(
        Object
            .entries((
                tableName !== undefined
                    ? initFromStorage(tableName, columns)
                    : columns
            ))
            .map(([name, visible]: [string, boolean]) => (
                [name, `${visible ? 'min' : 'max'}_mobileP`]
            )),
    )
);
