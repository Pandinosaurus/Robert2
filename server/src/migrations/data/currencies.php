<?php
declare(strict_types=1);

return [
    // ------------------------------------------------------
    // -
    // -    Francophones
    // -
    // ------------------------------------------------------

    ['code' => 'EUR', 'name' => 'Euro'],
    ['code' => 'CHF', 'name' => 'Franc suisse'],
    ['code' => 'CAD', 'name' => 'Dollar canadien'],
    ['code' => 'BIF', 'name' => 'Franc burundais'],
    ['code' => 'CDF', 'name' => 'Franc congolais'],
    ['code' => 'DJF', 'name' => 'Franc djiboutien'],
    ['code' => 'DZD', 'name' => 'Dinar algérien'],
    ['code' => 'GNF', 'name' => 'Franc guinéen'],
    ['code' => 'KMF', 'name' => 'Franc comorien'],
    ['code' => 'MAD', 'name' => 'Dirham marocain'],
    ['code' => 'RWF', 'name' => 'Franc rwandais'],
    ['code' => 'TND', 'name' => 'Dinar tunisien'],
    ['code' => 'XAF', 'name' => 'Franc CFA (BEAC)'],
    ['code' => 'XOF', 'name' => 'Franc CFA (BCEAO)'],
    ['code' => 'XPF', 'name' => 'Franc pacifique'],

    // ------------------------------------------------------
    // -
    // -    Non francophones
    // -
    // ------------------------------------------------------

    ['code' => 'AFN', 'name' => "Afghani afghan"],
    ['code' => 'MGA', 'name' => "Ariary malgache"],
    ['code' => 'THB', 'name' => "Baht thaïlandais"],
    ['code' => 'PAB', 'name' => "Balboa panaméen"],
    ['code' => 'ETB', 'name' => "Birr éthiopien"],
    ['code' => 'VES', 'name' => "Bolívar vénézuélien"],
    ['code' => 'BOB', 'name' => "Boliviano bolivien"],
    ['code' => 'GHS', 'name' => "Cedi ghanéen"],
    ['code' => 'CRC', 'name' => "Colón costaricien"],
    ['code' => 'NIO', 'name' => "Córdoba nicaraguayen"],
    ['code' => 'DKK', 'name' => "Couronne danoise"],
    ['code' => 'NOK', 'name' => "Couronne norvégienne"],
    ['code' => 'SEK', 'name' => "Couronne suédoise"],
    ['code' => 'CZK', 'name' => "Couronne tchèque"],
    ['code' => 'BHD', 'name' => "Dinar bahreïni"],
    ['code' => 'IQD', 'name' => "Dinar irakien"],
    ['code' => 'JOD', 'name' => "Dinar jordanien"],
    ['code' => 'KWD', 'name' => "Dinar koweïtien"],
    ['code' => 'LYD', 'name' => "Dinar libyen"],
    ['code' => 'MKD', 'name' => "Dinar macédonien"],
    ['code' => 'RSD', 'name' => "Dinar serbe"],
    ['code' => 'AED', 'name' => "Dirham des Émirats arabes unis"],
    ['code' => 'USD', 'name' => "Dollar américain"],
    ['code' => 'AUD', 'name' => "Dollar australien"],
    ['code' => 'BZD', 'name' => "Dollar bélizien"],
    ['code' => 'BND', 'name' => "Dollar brunéien"],
    ['code' => 'HKD', 'name' => "Dollar de Hong Kong"],
    ['code' => 'SGD', 'name' => "Dollar de Singapour"],
    ['code' => 'TWD', 'name' => "Dollar de Taïwan"],
    ['code' => 'TTD', 'name' => "Dollar de Trinité-et-Tobago"],
    ['code' => 'JMD', 'name' => "Dollar jamaïcain"],
    ['code' => 'NAD', 'name' => "Dollar namibien"],
    ['code' => 'NZD', 'name' => "Dollar néo-zélandais"],
    ['code' => 'ZWL', 'name' => "Dollar zimbabwéen"],
    ['code' => 'VND', 'name' => "Dong vietnamien"],
    ['code' => 'AMD', 'name' => "Dram arménien"],
    ['code' => 'CVE', 'name' => "Escudo cap-verdien"],
    ['code' => 'HUF', 'name' => "Forint hongrois"],
    ['code' => 'PYG', 'name' => "Guarani paraguayen"],
    ['code' => 'UAH', 'name' => "Hryvnia ukrainienne"],
    ['code' => 'ISK', 'name' => "Króna islandaise"],
    ['code' => 'ZMW', 'name' => "Kwacha zambien"],
    ['code' => 'MMK', 'name' => "Kyat birman"],
    ['code' => 'GEL', 'name' => "Lari géorgien"],
    ['code' => 'ALL', 'name' => "Lek albanais"],
    ['code' => 'HNL', 'name' => "Lempira hondurien"],
    ['code' => 'MDL', 'name' => "Leu moldave"],
    ['code' => 'RON', 'name' => "Leu roumain"],
    ['code' => 'BGN', 'name' => "Lev bulgare"],
    ['code' => 'EGP', 'name' => "Livre égyptienne"],
    ['code' => 'LBP', 'name' => "Livre libanaise"],
    ['code' => 'SDG', 'name' => "Livre soudanaise"],
    ['code' => 'GBP', 'name' => "Livre sterling"],
    ['code' => 'SYP', 'name' => "Livre syrienne"],
    ['code' => 'TRY', 'name' => "Livre turque"],
    ['code' => 'AZN', 'name' => "Manat azerbaïdjanais"],
    ['code' => 'BAM', 'name' => "Mark convertible bosniaque"],
    ['code' => 'MZN', 'name' => "Metical mozambicain"],
    ['code' => 'NGN', 'name' => "Naira nigérian"],
    ['code' => 'ERN', 'name' => "Nakfa érythréenne"],
    ['code' => 'PEN', 'name' => "Nuevo sol péruvien"],
    ['code' => 'TOP', 'name' => "Paʻanga tongien"],
    ['code' => 'MOP', 'name' => "Pataca macanaise"],
    ['code' => 'ARS', 'name' => "Peso argentin"],
    ['code' => 'CLP', 'name' => "Peso chilien"],
    ['code' => 'COP', 'name' => "Peso colombien"],
    ['code' => 'DOP', 'name' => "Peso dominicain"],
    ['code' => 'MXN', 'name' => "Peso mexicain"],
    ['code' => 'PHP', 'name' => "Peso philippin"],
    ['code' => 'UYU', 'name' => "Peso uruguayen"],
    ['code' => 'BWP', 'name' => "Pula botswanais"],
    ['code' => 'GTQ', 'name' => "Quetzal guatémaltèque"],
    ['code' => 'ZAR', 'name' => "Rand sud-africain"],
    ['code' => 'BRL', 'name' => "Real brésilien"],
    ['code' => 'IRR', 'name' => "Rial iranien"],
    ['code' => 'OMR', 'name' => "Rial omanais"],
    ['code' => 'QAR', 'name' => "Rial qatari"],
    ['code' => 'YER', 'name' => "Rial yéménite"],
    ['code' => 'KHR', 'name' => "Riel cambodgien"],
    ['code' => 'MYR', 'name' => "Ringgit malaisien"],
    ['code' => 'SAR', 'name' => "Riyal saoudien"],
    ['code' => 'BYN', 'name' => "Rouble biélorusse"],
    ['code' => 'RUB', 'name' => "Rouble russe"],
    ['code' => 'INR', 'name' => "Roupie indienne"],
    ['code' => 'MUR', 'name' => "Roupie mauricienne"],
    ['code' => 'NPR', 'name' => "Roupie népalaise"],
    ['code' => 'PKR', 'name' => "Roupie pakistanaise"],
    ['code' => 'LKR', 'name' => "Roupie srilankaise"],
    ['code' => 'IDR', 'name' => "Rupiah indonésienne"],
    ['code' => 'ILS', 'name' => "Shekel israélien"],
    ['code' => 'KES', 'name' => "Shilling kényan"],
    ['code' => 'UGX', 'name' => "Shilling ougandais"],
    ['code' => 'SOS', 'name' => "Shilling somalien"],
    ['code' => 'TZS', 'name' => "Shilling tanzanien"],
    ['code' => 'UZS', 'name' => "Som ouzbek"],
    ['code' => 'BDT', 'name' => "Taka bangladais"],
    ['code' => 'KZT', 'name' => "Tenge kazakh"],
    ['code' => 'KRW', 'name' => "Won sud-coréen"],
    ['code' => 'JPY', 'name' => "Yen japonais"],
    ['code' => 'CNY', 'name' => "Yuan chinois"],
    ['code' => 'PLN', 'name' => "Zloty polonais"],
];
