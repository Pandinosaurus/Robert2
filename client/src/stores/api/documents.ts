import { z } from '@/utils/validation';
import requester from '@/globals/requester';

import type { SchemaInfer } from '@/utils/validation';

// ------------------------------------------------------
// -
// -    Schema / Enums
// -
// ------------------------------------------------------

export const DocumentSchema = z.strictObject({
    id: z.number(),
    name: z.string(),
    type: z.string(),
    size: z.number(),
    url: z.string(),
    created_at: z.datetime(),
});

// ------------------------------------------------------
// -
// -    Types
// -
// ------------------------------------------------------

export type Document = SchemaInfer<typeof DocumentSchema>;

// ------------------------------------------------------
// -
// -    Fonctions
// -
// ------------------------------------------------------

const remove = async (id: Document['id']): Promise<void> => {
    await requester.delete(`/documents/${id}`);
};

export default { remove };
