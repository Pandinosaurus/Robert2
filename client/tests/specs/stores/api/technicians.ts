import requester from '@/globals/requester';
import apiTechnicians from '@/stores/api/technicians';
import { withPaginationEnvelope } from '@fixtures/@utils';
import documents from '@fixtures/documents';
import data from '@fixtures/technicians';

describe('Technicians Api', () => {
    describe('all()', () => {
        it('parse the returned data correctly', async () => {
            const paginatedData = withPaginationEnvelope(data.default());
            jest.spyOn(requester, 'get').mockResolvedValue({ data: paginatedData });
            expect(await apiTechnicians.all()).toMatchSnapshot();
        });
    });

    describe('allWhileEvent()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data: data.withEvents() });
            expect(await apiTechnicians.allWhileEvent(1)).toMatchSnapshot();
        });
    });

    describe('one()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data: datum });
            expect(await apiTechnicians.one(datum.id)).toMatchSnapshot();
        });
    });

    describe('create()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'post').mockResolvedValue({ data: datum });
            expect(await apiTechnicians.create({} as any)).toMatchSnapshot();
        });
    });

    describe('update()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
            expect(await apiTechnicians.update(datum.id, {} as any)).toMatchSnapshot();
        });
    });

    describe('documents()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data: documents.default() });
            expect(await apiTechnicians.documents(1)).toMatchSnapshot();
        });
    });

    describe('assignments()', () => {
        it.each(data.withEvents())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data: datum.events });
            expect(await apiTechnicians.assignments(datum.id)).toMatchSnapshot();
        });
    });

    describe('attachDocument()', () => {
        it.each(documents.default())('parse the returned data correctly (with document #$id)', async (datum: any) => {
            jest.spyOn(requester, 'post').mockResolvedValue({ data: datum });
            const fakeFile = new File(['__TEST__'], 'my-document.txt', { type: 'text/plain' });
            expect(await apiTechnicians.attachDocument(1, fakeFile)).toMatchSnapshot();
        });
    });

    describe('restore()', () => {
        it.each(data.details())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
            expect(await apiTechnicians.restore(datum.id)).toMatchSnapshot();
        });
    });
});
