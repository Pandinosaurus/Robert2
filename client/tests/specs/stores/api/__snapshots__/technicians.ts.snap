// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technicians Api all() parse the returned data correctly 1`] = `
{
  "data": [
    {
      "country": null,
      "country_id": null,
      "email": "tester2@robertmanager.net",
      "first_name": "Roger",
      "full_address": null,
      "full_name": "Roger Rabbit",
      "id": 1,
      "last_name": "Rabbit",
      "locality": null,
      "nickname": "Riri",
      "note": null,
      "phone": null,
      "postal_code": null,
      "street": null,
      "user_id": 2,
    },
    {
      "country": {
        "code": "CH",
        "id": 2,
        "name": "Suisse",
      },
      "country_id": 2,
      "email": "client@technicien.com",
      "first_name": "Jean",
      "full_address": null,
      "full_name": "Jean Technicien",
      "id": 2,
      "last_name": "Technicien",
      "locality": null,
      "nickname": null,
      "note": null,
      "phone": "+33645698520",
      "postal_code": null,
      "street": null,
      "user_id": null,
    },
  ],
  "pagination": {
    "currentPage": 1,
    "perPage": 2,
    "total": {
      "items": 2,
      "pages": 1,
    },
  },
}
`;

exports[`Technicians Api allWhileEvent() parse the returned data correctly 1`] = `
[
  {
    "country": null,
    "country_id": null,
    "email": "tester2@robertmanager.net",
    "events": [
      {
        "event": {
          "color": null,
          "created_at": /* DateTime Instance => */ "2018-12-01 12:50:45",
          "description": null,
          "id": 1,
          "is_archived": false,
          "is_billable": true,
          "is_confirmed": false,
          "is_departure_inventory_done": true,
          "is_return_inventory_done": true,
          "location": "Gap",
          "materials_count": 3,
          "mobilization_period": /* Period Instance => */ {
            "end": "2018-12-19 00:00:00",
            "isFullDays": false,
            "start": "2018-12-16 15:45:00",
          },
          "note": null,
          "operation_period": /* Period Instance => */ {
            "end": "2018-12-18 18:00:00",
            "isFullDays": false,
            "start": "2018-12-17 10:00:00",
          },
          "reference": null,
          "title": "Premier événement",
          "updated_at": /* DateTime Instance => */ "2018-12-05 08:31:21",
        },
        "event_id": 1,
        "id": 1,
        "period": /* Period Instance => */ {
          "end": "2018-12-18 22:00:00",
          "isFullDays": false,
          "start": "2018-12-17 09:00:00",
        },
        "position": "Régisseur",
        "technician_id": 1,
      },
    ],
    "first_name": "Roger",
    "full_address": null,
    "full_name": "Roger Rabbit",
    "id": 1,
    "last_name": "Rabbit",
    "locality": null,
    "nickname": "Riri",
    "note": null,
    "phone": null,
    "postal_code": null,
    "street": null,
    "user_id": 2,
  },
  {
    "country": {
      "code": "CH",
      "id": 2,
      "name": "Suisse",
    },
    "country_id": 2,
    "email": "client@technicien.com",
    "events": [
      {
        "event": {
          "color": null,
          "created_at": /* DateTime Instance => */ "2018-12-01 12:50:45",
          "description": null,
          "id": 1,
          "is_archived": false,
          "is_billable": true,
          "is_confirmed": false,
          "is_departure_inventory_done": true,
          "is_return_inventory_done": true,
          "location": "Gap",
          "materials_count": 3,
          "mobilization_period": /* Period Instance => */ {
            "end": "2018-12-19 00:00:00",
            "isFullDays": false,
            "start": "2018-12-16 15:45:00",
          },
          "note": null,
          "operation_period": /* Period Instance => */ {
            "end": "2018-12-18 18:00:00",
            "isFullDays": false,
            "start": "2018-12-17 10:00:00",
          },
          "reference": null,
          "title": "Premier événement",
          "updated_at": /* DateTime Instance => */ "2018-12-05 08:31:21",
        },
        "event_id": 1,
        "id": 2,
        "period": /* Period Instance => */ {
          "end": "2018-12-18 18:00:00",
          "isFullDays": false,
          "start": "2018-12-18 14:00:00",
        },
        "position": "Technicien plateau",
        "technician_id": 2,
      },
      {
        "event": {
          "color": null,
          "created_at": /* DateTime Instance => */ "2022-05-29 18:00:00",
          "description": null,
          "id": 7,
          "is_archived": false,
          "is_billable": true,
          "is_confirmed": true,
          "is_departure_inventory_done": false,
          "is_return_inventory_done": false,
          "location": "Machin-le-chateau",
          "materials_count": 7,
          "mobilization_period": /* Period Instance => */ {
            "end": "2023-05-29 00:00:00",
            "isFullDays": false,
            "start": "2023-05-25 00:00:00",
          },
          "note": null,
          "operation_period": /* Period Instance => */ {
            "end": "2023-05-28",
            "isFullDays": true,
            "start": "2023-05-25",
          },
          "reference": null,
          "title": "Médiévales de Machin-le-chateau 2023",
          "updated_at": /* DateTime Instance => */ "2022-05-29 18:05:00",
        },
        "event_id": 7,
        "id": 3,
        "period": /* Period Instance => */ {
          "end": "2023-05-29 00:00:00",
          "isFullDays": false,
          "start": "2023-05-25 00:00:00",
        },
        "position": "Ingénieur du son",
        "technician_id": 2,
      },
    ],
    "first_name": "Jean",
    "full_address": null,
    "full_name": "Jean Technicien",
    "id": 2,
    "last_name": "Technicien",
    "locality": null,
    "nickname": null,
    "note": null,
    "phone": "+33645698520",
    "postal_code": null,
    "street": null,
    "user_id": null,
  },
]
`;

exports[`Technicians Api assignments() parse the returned data correctly (with #1) 1`] = `
[
  {
    "event": {
      "color": null,
      "created_at": /* DateTime Instance => */ "2018-12-01 12:50:45",
      "description": null,
      "id": 1,
      "is_archived": false,
      "is_billable": true,
      "is_confirmed": false,
      "is_departure_inventory_done": true,
      "is_return_inventory_done": true,
      "location": "Gap",
      "materials_count": 3,
      "mobilization_period": /* Period Instance => */ {
        "end": "2018-12-19 00:00:00",
        "isFullDays": false,
        "start": "2018-12-16 15:45:00",
      },
      "note": null,
      "operation_period": /* Period Instance => */ {
        "end": "2018-12-18 18:00:00",
        "isFullDays": false,
        "start": "2018-12-17 10:00:00",
      },
      "reference": null,
      "title": "Premier événement",
      "updated_at": /* DateTime Instance => */ "2018-12-05 08:31:21",
    },
    "event_id": 1,
    "id": 1,
    "period": /* Period Instance => */ {
      "end": "2018-12-18 22:00:00",
      "isFullDays": false,
      "start": "2018-12-17 09:00:00",
    },
    "position": "Régisseur",
    "technician_id": 1,
  },
]
`;

exports[`Technicians Api assignments() parse the returned data correctly (with #2) 1`] = `
[
  {
    "event": {
      "color": null,
      "created_at": /* DateTime Instance => */ "2018-12-01 12:50:45",
      "description": null,
      "id": 1,
      "is_archived": false,
      "is_billable": true,
      "is_confirmed": false,
      "is_departure_inventory_done": true,
      "is_return_inventory_done": true,
      "location": "Gap",
      "materials_count": 3,
      "mobilization_period": /* Period Instance => */ {
        "end": "2018-12-19 00:00:00",
        "isFullDays": false,
        "start": "2018-12-16 15:45:00",
      },
      "note": null,
      "operation_period": /* Period Instance => */ {
        "end": "2018-12-18 18:00:00",
        "isFullDays": false,
        "start": "2018-12-17 10:00:00",
      },
      "reference": null,
      "title": "Premier événement",
      "updated_at": /* DateTime Instance => */ "2018-12-05 08:31:21",
    },
    "event_id": 1,
    "id": 2,
    "period": /* Period Instance => */ {
      "end": "2018-12-18 18:00:00",
      "isFullDays": false,
      "start": "2018-12-18 14:00:00",
    },
    "position": "Technicien plateau",
    "technician_id": 2,
  },
  {
    "event": {
      "color": null,
      "created_at": /* DateTime Instance => */ "2022-05-29 18:00:00",
      "description": null,
      "id": 7,
      "is_archived": false,
      "is_billable": true,
      "is_confirmed": true,
      "is_departure_inventory_done": false,
      "is_return_inventory_done": false,
      "location": "Machin-le-chateau",
      "materials_count": 7,
      "mobilization_period": /* Period Instance => */ {
        "end": "2023-05-29 00:00:00",
        "isFullDays": false,
        "start": "2023-05-25 00:00:00",
      },
      "note": null,
      "operation_period": /* Period Instance => */ {
        "end": "2023-05-28",
        "isFullDays": true,
        "start": "2023-05-25",
      },
      "reference": null,
      "title": "Médiévales de Machin-le-chateau 2023",
      "updated_at": /* DateTime Instance => */ "2022-05-29 18:05:00",
    },
    "event_id": 7,
    "id": 3,
    "period": /* Period Instance => */ {
      "end": "2023-05-29 00:00:00",
      "isFullDays": false,
      "start": "2023-05-25 00:00:00",
    },
    "position": "Ingénieur du son",
    "technician_id": 2,
  },
]
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #1) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2021-02-12 13:23:02",
  "id": 1,
  "name": "User-manual.pdf",
  "size": 24681233,
  "type": "application/pdf",
  "url": "http://loxya.test/documents/1",
}
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #2) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2021-02-12 13:25:02",
  "id": 2,
  "name": "warranty.pdf",
  "size": 124068,
  "type": "application/pdf",
  "url": "http://loxya.test/documents/2",
}
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #3) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
  "id": 3,
  "name": "plan-de-salle.xls",
  "size": 49802,
  "type": "application/vnd.ms-excel",
  "url": "http://loxya.test/documents/3",
}
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #4) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2023-05-02 17:32:05",
  "id": 4,
  "name": "Carte de l'étudiant.png",
  "size": 9397,
  "type": "image/x-png",
  "url": "http://loxya.test/documents/4",
}
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #5) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
  "id": 5,
  "name": "bon_de_sortie.doc",
  "size": 10014149,
  "type": "application/msword",
  "url": "http://loxya.test/documents/5",
}
`;

exports[`Technicians Api attachDocument() parse the returned data correctly (with document #6) 1`] = `
{
  "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
  "id": 6,
  "name": "emploi-du-temps.xls",
  "size": 70001,
  "type": "application/vnd.ms-excel",
  "url": "http://loxya.test/documents/6",
}
`;

exports[`Technicians Api create() parse the returned data correctly (with #1) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "id": 1,
  "last_name": "Rabbit",
  "locality": null,
  "nickname": "Riri",
  "note": null,
  "phone": null,
  "postal_code": null,
  "street": null,
  "user": {
    "email": "tester2@robertmanager.net",
    "first_name": "Roger",
    "full_name": "Roger Rabbit",
    "group": "management",
    "id": 2,
    "last_name": "Rabbit",
    "phone": null,
    "pseudo": "test2",
  },
  "user_id": 2,
}
`;

exports[`Technicians Api create() parse the returned data correctly (with #2) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "client@technicien.com",
  "first_name": "Jean",
  "full_address": null,
  "full_name": "Jean Technicien",
  "id": 2,
  "last_name": "Technicien",
  "locality": null,
  "nickname": null,
  "note": null,
  "phone": "+33645698520",
  "postal_code": null,
  "street": null,
  "user": null,
  "user_id": null,
}
`;

exports[`Technicians Api documents() parse the returned data correctly 1`] = `
[
  {
    "created_at": /* DateTime Instance => */ "2021-02-12 13:23:02",
    "id": 1,
    "name": "User-manual.pdf",
    "size": 24681233,
    "type": "application/pdf",
    "url": "http://loxya.test/documents/1",
  },
  {
    "created_at": /* DateTime Instance => */ "2021-02-12 13:25:02",
    "id": 2,
    "name": "warranty.pdf",
    "size": 124068,
    "type": "application/pdf",
    "url": "http://loxya.test/documents/2",
  },
  {
    "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
    "id": 3,
    "name": "plan-de-salle.xls",
    "size": 49802,
    "type": "application/vnd.ms-excel",
    "url": "http://loxya.test/documents/3",
  },
  {
    "created_at": /* DateTime Instance => */ "2023-05-02 17:32:05",
    "id": 4,
    "name": "Carte de l'étudiant.png",
    "size": 9397,
    "type": "image/x-png",
    "url": "http://loxya.test/documents/4",
  },
  {
    "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
    "id": 5,
    "name": "bon_de_sortie.doc",
    "size": 10014149,
    "type": "application/msword",
    "url": "http://loxya.test/documents/5",
  },
  {
    "created_at": /* DateTime Instance => */ "2023-05-01 15:15:20",
    "id": 6,
    "name": "emploi-du-temps.xls",
    "size": 70001,
    "type": "application/vnd.ms-excel",
    "url": "http://loxya.test/documents/6",
  },
]
`;

exports[`Technicians Api one() parse the returned data correctly (with #1) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "id": 1,
  "last_name": "Rabbit",
  "locality": null,
  "nickname": "Riri",
  "note": null,
  "phone": null,
  "postal_code": null,
  "street": null,
  "user": {
    "email": "tester2@robertmanager.net",
    "first_name": "Roger",
    "full_name": "Roger Rabbit",
    "group": "management",
    "id": 2,
    "last_name": "Rabbit",
    "phone": null,
    "pseudo": "test2",
  },
  "user_id": 2,
}
`;

exports[`Technicians Api one() parse the returned data correctly (with #2) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "client@technicien.com",
  "first_name": "Jean",
  "full_address": null,
  "full_name": "Jean Technicien",
  "id": 2,
  "last_name": "Technicien",
  "locality": null,
  "nickname": null,
  "note": null,
  "phone": "+33645698520",
  "postal_code": null,
  "street": null,
  "user": null,
  "user_id": null,
}
`;

exports[`Technicians Api restore() parse the returned data correctly (with #1) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "id": 1,
  "last_name": "Rabbit",
  "locality": null,
  "nickname": "Riri",
  "note": null,
  "phone": null,
  "postal_code": null,
  "street": null,
  "user": {
    "email": "tester2@robertmanager.net",
    "first_name": "Roger",
    "full_name": "Roger Rabbit",
    "group": "management",
    "id": 2,
    "last_name": "Rabbit",
    "phone": null,
    "pseudo": "test2",
  },
  "user_id": 2,
}
`;

exports[`Technicians Api restore() parse the returned data correctly (with #2) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "client@technicien.com",
  "first_name": "Jean",
  "full_address": null,
  "full_name": "Jean Technicien",
  "id": 2,
  "last_name": "Technicien",
  "locality": null,
  "nickname": null,
  "note": null,
  "phone": "+33645698520",
  "postal_code": null,
  "street": null,
  "user": null,
  "user_id": null,
}
`;

exports[`Technicians Api update() parse the returned data correctly (with #1) 1`] = `
{
  "country": null,
  "country_id": null,
  "email": "tester2@robertmanager.net",
  "first_name": "Roger",
  "full_address": null,
  "full_name": "Roger Rabbit",
  "id": 1,
  "last_name": "Rabbit",
  "locality": null,
  "nickname": "Riri",
  "note": null,
  "phone": null,
  "postal_code": null,
  "street": null,
  "user": {
    "email": "tester2@robertmanager.net",
    "first_name": "Roger",
    "full_name": "Roger Rabbit",
    "group": "management",
    "id": 2,
    "last_name": "Rabbit",
    "phone": null,
    "pseudo": "test2",
  },
  "user_id": 2,
}
`;

exports[`Technicians Api update() parse the returned data correctly (with #2) 1`] = `
{
  "country": {
    "code": "CH",
    "id": 2,
    "name": "Suisse",
  },
  "country_id": 2,
  "email": "client@technicien.com",
  "first_name": "Jean",
  "full_address": null,
  "full_name": "Jean Technicien",
  "id": 2,
  "last_name": "Technicien",
  "locality": null,
  "nickname": null,
  "note": null,
  "phone": "+33645698520",
  "postal_code": null,
  "street": null,
  "user": null,
  "user_id": null,
}
`;
